Improve Daml installation documentation for "No space left" error (#21350)
digital-asset/daml: sdk/docs/manually-written/sdk/tutorials/setup-assistant/installation.rst

1. `Visual Studio Code <https://code.visualstudio.com/download>`_.
2. JDK 11 or greater. If you don't already have a JDK installed, try `Eclipse Adoptium <https://adoptium.net>`_.

>    As part of the installation process you may need to set up the ``JAVA_HOME`` variable. You can find instructions for this process on :doc:`Windows,macOS, and Linux in the :ref:`path-variables <path-variables>` section.

Reviewer: I don't think the `ref` is necessary here, since we're already inside of a :doc: directive this probably breaks compilation.

If you want to verify the SDK download for security purposes before installing, you can look at :doc:`our detailed instructions for manual download and installation <manual-download>`.

Troubleshooting
***************

> The Daml installation requires x amount of disk space. If there is not enough space on disk, the installer outputs the error:

Author: "x" here should be replaced with some measure of the total space required, perhaps the reviewers have an idea?

Reviewer 2: I've had a quick look at the SDK installations I have on my machine:
2.10 -> 1.6 GB
3.3.0 -> 1.1 GB

I'd suggest to write `around 1.5 GB` of disk space.

``
resource exhausted (No space left on device)
``

In order to fix this problem, one needs to ensure sufficient disk space overall, and in particuloar that the user's home directory and the temporary
files directory have sufficient space (i.e., in the order of 2GiB each):

* On Windows, one can find the user's home directory in the `%USERPROFILE%` environment variable, and the temporary files directory in the
  ``%TEMP%`` environment variable
* On Linux and MacOS, one can find the user's home directory in the ``HOME%`` environment variable, and the temporary files directory in the
  ``TMPDIR`` environment variable 

If the installer cannot ensure sufficient space in the the user's home directory and the temporary files directory, an alternative is to change
the location of these directories before the installation.

> To do so, please set the ``DAML_HOME`` and ``TMPDIR`` environment variables to other directories that have sufficient space. You can find instructions on how to do this :doc:`here <path-variables>`.

Reviewer 2: please avoid "here" references as per style-guide: https://docs.google.com/document/d/153D6-gI0blsAvlrzReHVyz8RCHUQzzdKnxnpcADlt6c/edit?tab=t.0#bookmark=id.ya2a6pwloo5b. Also a `:ref:` is preferred, as it is more stable under doc refactorings than a `:doc:` reference.

> To do so, please set the ``DAML_HOME`` and ``TMPDIR`` environment variables to other directories that have sufficient space. You can find instructions on how to do this :doc:`here <path-variables>`.

Reviewer 2: Any chance to inline the command line:
```
DAML_HOME=<YOUR-TARGET-DIR> TMPDIR=<YOUR-TEMPDIR> curl -sSL https://get.daml.com/ | sh
```

> To do so, please set the ``DAML_HOME`` and ``TMPDIR`` environment variables to other directories that have sufficient space. You can find instructions on how to do this :doc:`here <path-variables>`.

Reviewer 2: ```suggestion
To do so, call the installer with ``DAML_HOME`` and ``TMPDIR`` environment variables pointing to directories that have sufficient space:

.. code-block: shell
  
  DAML_HOME=<YOUR-TARGET-DIR> TMPDIR=<YOUR-TEMPDIR> curl -sSL https://get.daml.com/ | sh
```

(Not sure about the code-block syntax.)

Author: Done. I've added slightly different instructions for Windows, which should be fine, but could you please take another look?

> It is important that the ``DAML_HOME`` and ``TMPDIR`` environment variables do not point to directories located inside the unzipped installation

Reviewer 2: ```suggestion
Avoid choosing directories for ``DAML_HOME`` and ``TMPDIR`` environment variables that are located inside the unzipped installation
```

> direcory ``sdk-x.x.x``. If they do, the installation does not complete.

Reviewer 2: ```suggestion
direcory ``sdk-x.x.x``, as otherwise the installer fails.
```

If you want to verify the SDK download for security purposes before installing, you can look at :doc:`our detailed instructions for manual download and installation <manual-download>`.

Troubleshooting
***************

The Daml installation requires around 1.5 GB of disk space. If there is not enough space on disk, the installer outputs the error:
``
resource exhausted (No space left on device)
``

In order to fix this problem, one needs to ensure sufficient disk space overall, and in particuloar that the user's home directory and the temporary
files directory have sufficient space (i.e., in the order of 2GiB each):

> * On Windows, one can find the user's home directory in the `%USERPROFILE%` environment variable, and the temporary files directory in the

Reviewer: I think we need to mention DAML_HOME instead of HOME here, since HOME is only partially relevant for where the SDK installs.

According to our code:

```
-- | The DAML_HOME environment variable determines the path of the daml
-- assistant data directory. This defaults to:
--
--     System.Directory.getAppUserDataDirectory "daml"
--
-- On Linux and Mac, that's ~/.daml
-- On Windows, that's %APPDATA%/daml
```

So we could tell the user that DAML_HOME sets where Daml SDK is installed, and that it defaults to the locations listed above.

  ``%TEMP%`` environment variable

> * On Linux and MacOS, one can find the user's home directory in the ``HOME%`` environment variable, and the temporary files directory in the

Reviewer: I don't think that `%` symbol is accurate, no?
```suggestion
* On Linux and MacOS, one can find the user's home directory in the ``HOME`` environment variable, and the temporary files directory in the
```

  ``TMPDIR`` environment variable 

> If the installer cannot ensure sufficient space in the the user's home directory and the temporary files directory, an alternative is to change

Reviewer: ```suggestion
If the installer cannot ensure sufficient space in the Daml SDK's home directory (``DAML_HOME``) and the temporary files directory, an alternative is to change
```

Depending on how you word the section before this one, just saying the name of the environment variable may be more appropriate.

the location of these directories before the installation.
To do so, call the installer with ``DAML_HOME`` and ``TMPDIR`` environment variables pointing to directories that have sufficient space:

* On Windows, refer to the :ref:`path-variables <path-variables>` section for setting the environment variables. Then run the installation as usual.
* On Linus and MacOS, run in a terminal:
 
.. code-block: bash
  

>   DAML_HOME=<YOUR-TARGET-DIR> TMPDIR=<YOUR-TEMPDIR> curl -sSL https://get.daml.com/ | sh

Reviewer: We should also mention that you'll need to have the same DAML_HOME set and add `$DAML_HOME/bin` to the path in all terminals you use in the future, probably in .bashrc. Up to you if you want to include this here or in path-variables.rst

Avoid choosing directories for ``DAML_HOME`` and ``TMPDIR`` environment variables that are located inside the unzipped installation
directory ``sdk-x.x.x``, as otherwise the installer fails.

> 

Reviewer: Recommending DAML_HOME also requires that we update the docs for path-variables.rst, since there we tell people to just use commands like
```
        echo 'export PATH="$HOME/.daml/bin:$PATH"' >> ~/.bash_profile
```
which won't be correct when DAML_HOME is set.
